src/workers/newNotificationV2RealTime.ts

```diff
@@ -0,0 +1,55 @@
+import { messageToJson, Worker } from './worker';
+import { redisPubSub } from '../redis';
+import { ChangeObject } from '../types';
+import {
+  getNotificationV2AndChildren,
+  NotificationV2,
+  streamNotificationUsers,
+} from '../entity/notifications/NotificationV2';
+import { processStream } from '../common/streaming';
+
+interface Data {
+  notification: ChangeObject<NotificationV2>;
+}
+
+const QUEUE_CONCURRENCY = 10;
+
+const worker: Worker = {
+  subscription: 'api.new-notification-real-time',
+  handler: async (message, con, logger): Promise<void> => {
+    const data: Data = messageToJson(message);
+    if (!data.notification.public) {
+      return;
+    }
+    try {
+      const { id } = data.notification;
+      const [notification, attachments, avatars] =
+        await getNotificationV2AndChildren(con, id);
+      if (notification) {
+        const stream = await streamNotificationUsers(con, notification.id);
+        await processStream(
+          stream,
+          ({ userId }) =>
+            redisPubSub.publish(`events.notifications.${userId}.new`, {
+              ...notification,
+              attachments,
+              avatars,
+            }),
+          QUEUE_CONCURRENCY,
+        );
+      }
+    } catch (err) {
+      // Don't throw the error because we send the notification to multiple users, retrying may spam some users
```

Comment:
Would this give us issues down the line if it errors often basically means people will not get realtime?